File: PostmanTest/EnclosureTests/AttachedEnclosureTest.vb
Imports System.Net.Mail
Imports Postman

'''<summary>
'''This is a test class for AttachedEnclosure and is intended to contain all AttachedEnclosure Unit Tests
'''</summary>
<TestClass()> _
Public Class AttachedEnclosureTest

    '''<summary>
    '''A test for AttachedEnclosure Constructor
    '''</summary>
    <TestMethod()> _
    Public Sub AttachedEnclosureConstructorTest()
        'Arrange
        Dim expectedAttachementCount As Integer = 1
        Dim expectedAttachment As New Attachment(New System.IO.MemoryStream(), "Test Attachement")
        Dim msg As New MailMessage()
        Dim target As IEnclosure = New Enclosure.Attached(expectedAttachment)

        'Act
        target.Include(msg)

        'Assert
        Assert.AreEqual(expectedAttachementCount, msg.Attachments.Count)
        Assert.AreEqual(expectedAttachment.Name, msg.Attachments(0).Name)
    End Sub

End Class

File: Postman.Tests/Enclosure/AttachedEnclosureTest.cs
namespace Postman.Tests
{
    using System.Net.Mail;
    using Postman.Interfaces;
    using Xunit;

    /// <summary>
    /// This is a test class for AttachedEnclosure and is intended to contain all AttachedEnclosure Unit Tests
    /// </summary>
    public class AttachedEnclosureTest
    {
        /// <summary>
        /// A test for AttachedEnclosure Constructor
        /// </summary>
        [Fact]
        public void AttachedEnclosureConstructorTest()
        {
            // Arrange
            int expectedAttachementCount = 1;
            Attachment expectedAttachment = new Attachment(new System.IO.MemoryStream(), "Test Attachement");
            MailMessage msg = new MailMessage();
            IEnclosure target = new Enclosure.Attached(expectedAttachment);

            // Act
            target.Include(msg);

            // Assert
            Assert.Equal(expectedAttachementCount, msg.Attachments.Count);
            Assert.Equal(expectedAttachment.Name, msg.Attachments[0].Name);
        }
    }
}